code/go/0chain.net/blobbercore/reference/ref.go

```diff
@@ -56,40 +56,40 @@ func (a *Attributes) Validate() (err error) {
 }
 
 type Ref struct {
-	ID                  int64          `gorm:"column:id;primary_key"`
-	Type                string         `gorm:"column:type" dirlist:"type" filelist:"type"`
-	AllocationID        string         `gorm:"column:allocation_id"`
-	LookupHash          string         `gorm:"column:lookup_hash" dirlist:"lookup_hash" filelist:"lookup_hash"`
-	Name                string         `gorm:"column:name" dirlist:"name" filelist:"name"`
-	Path                string         `gorm:"column:path" dirlist:"path" filelist:"path"`
-	Hash                string         `gorm:"column:hash" dirlist:"hash" filelist:"hash"`
-	NumBlocks           int64          `gorm:"column:num_of_blocks" dirlist:"num_of_blocks" filelist:"num_of_blocks"`
-	PathHash            string         `gorm:"column:path_hash" dirlist:"path_hash" filelist:"path_hash"`
-	ParentPath          string         `gorm:"column:parent_path"`
-	PathLevel           int            `gorm:"column:level"`
-	CustomMeta          string         `gorm:"column:custom_meta" filelist:"custom_meta"`
-	ContentHash         string         `gorm:"column:content_hash" filelist:"content_hash"`
-	Size                int64          `gorm:"column:size" dirlist:"size" filelist:"size"`
-	MerkleRoot          string         `gorm:"column:merkle_root" filelist:"merkle_root"`
-	ActualFileSize      int64          `gorm:"column:actual_file_size" filelist:"actual_file_size"`
-	ActualFileHash      string         `gorm:"column:actual_file_hash" filelist:"actual_file_hash"`
-	MimeType            string         `gorm:"column:mimetype" filelist:"mimetype"`
-	WriteMarker         string         `gorm:"column:write_marker"`
-	ThumbnailSize       int64          `gorm:"column:thumbnail_size" filelist:"thumbnail_size"`
-	ThumbnailHash       string         `gorm:"column:thumbnail_hash" filelist:"thumbnail_hash"`
-	ActualThumbnailSize int64          `gorm:"column:actual_thumbnail_size" filelist:"actual_thumbnail_size"`
-	ActualThumbnailHash string         `gorm:"column:actual_thumbnail_hash" filelist:"actual_thumbnail_hash"`
-	EncryptedKey        string         `gorm:"column:encrypted_key" filelist:"encrypted_key"`
-	Attributes          datatypes.JSON `gorm:"column:attributes" filelist:"attributes"`
-	Children            []*Ref         `gorm:"-"`
-	childrenLoaded      bool
-
-	OnCloud        bool            `gorm:"column:on_cloud" filelist:"on_cloud"`
-	CommitMetaTxns []CommitMetaTxn `gorm:"foreignkey:ref_id" filelist:"commit_meta_txns"`
-	CreatedAt      time.Time       `gorm:"column:created_at" dirlist:"created_at" filelist:"created_at"`
-	UpdatedAt      time.Time       `gorm:"column:updated_at" dirlist:"updated_at" filelist:"updated_at"`
-
-	DeletedAt gorm.DeletedAt `gorm:"column:deleted_at"` // soft deletion
+	ID                  int64          `gorm:"column:id;primary_key" json:"id,omitempty"`
```

Comment:
Why add json annotations explicitly? Gorm handles the json naming imo.

Reply:
It gives flexibility and also I am not sure about Gorm handling json naming. It supports json dataTypes but it always fills struct fields with its value and its easy to use it with json tag.

Reply:
json column field is used when we unmarshal an interface to the following dataType, but since we already have gorm annotation, gorm has default serialising strategy as `snake_case` hence I said this change is redundant.